### PR TITLE
MySQL Join implementation

### PR DESCRIPTION
**Pull request by bajinsheng**
Part of #918

### CODE REVIEW

File: src/sqlancer/mysql/MySQLProvider.java [hunk 1]
```diff
@@ -148,7 +140,7 @@ private static int mapActions(MySQLGlobalState globalState, Action a) {
 
     @Override
     public void generateDatabase(MySQLGlobalState globalState) throws Exception {
-        while (globalState.getSchema().getDatabaseTables().size() < Randomly.smallNumber() + 1) {
+        while (globalState.getSchema().getDatabaseTables().size() < Randomly.getNotCachedInteger(1, 6)) {
```
Comment (mrigger) [inline src/sqlancer/mysql/MySQLProvider.java:143]: 6 tables seem like a lot here and might have a detrimental effect on performance. Could we perhaps set the limit depending on the test oracle here?

File: src/sqlancer/mysql/MySQLProvider.java [hunk 2]
```diff
@@ -148,7 +140,7 @@ private static int mapActions(MySQLGlobalState globalState, Action a) {
 
     @Override
     public void generateDatabase(MySQLGlobalState globalState) throws Exception {
-        while (globalState.getSchema().getDatabaseTables().size() < Randomly.smallNumber() + 1) {
+        while (globalState.getSchema().getDatabaseTables().size() < Randomly.getNotCachedInteger(1, 2)) {
```
Comment (mrigger) [inline src/sqlancer/mysql/MySQLProvider.java:143]: I think we could just leave `Randomly.smallNumber() + 1`, which might sometimes generate more tables than 2?

Comment (bajinsheng) [reply src/sqlancer/mysql/MySQLProvider.java:143]: No. Small number returns (-1, 1) with a Gaussian distribution, which means in most situations, there is only one table.

Comment (mrigger) [reply src/sqlancer/mysql/MySQLProvider.java:143]: Thanks for the clarification!

**Pull request merged**